src/aap_eda/tasks/ruleset.py

```diff
@@ -158,6 +176,18 @@ def restart(activation_id: int, requester: str = "User") -> None:
         _schedule_activate(activation, requester)
 
 
+def _job_in_queue(queue: Queue, job_id: str) -> Optional[Job]:
```

Comment:
This function and your approach to make a task unique is something that is going to be reusable. At least I would follow the pattern for https://github.com/ansible/eda-server/pull/396. 

Can be this function public and located in a common place?